### PR TITLE
feat: use virtual select UI in role edit page

### PR DESCRIPTION
**Pull request by ljl66-66**
Fix: https://github.com/casdoor/casdoor/issues/3316

using virtual list technology,more data is dynamically loaded. let variable virtual=true

### CODE REVIEW

**Comment by ljl66-66** [comment]
In fact, there is another solution that I have tested: similar to lazy loading technology. When entering the character editing page, get a small part of the data by pagination, and then click the drop-down selection box, and then load the remaining data, which can fundamentally solve the problem of slow loading into the character editing page. But after my test, this solution is not effectively compatible with the data before entering the editing page, and after entering the editing page, if you want to get more data in the drop-down list, you will continue to call the backend API, which will also have time loss, affecting the user experience, and also increasing the burden on the backend server. Therefore, I did not adopt this solution.

**Review by dacongda — APPROVED** [review]

**Comment by dacongda** [comment]
in face, i think those new changes is match the theme of this pr, plz make a new pr to implement the new feather

**Comment by ljl66-66** [comment]
> in face, i think those new changes is not match the theme of this pr, plz make a new pr to implement the new feather

How is it now?

File: web/src/RoleEditPage.js [hunk 1]
```diff
@@ -187,7 +187,7 @@ class RoleEditPage extends React.Component {
             {Setting.getLabel(i18next.t("role:Sub users"), i18next.t("role:Sub users - Tooltip"))} :
           </Col>
           <Col span={22} >
-            <Select virtual={false} mode="multiple" style={{width: "100%"}} value={this.state.role.users}
+            <Select virtual={true} mode="multiple" style={{width: "100%"}} value={this.state.role.users}
```
Comment (dacongda) [inline web/src/RoleEditPage.js:190]: remove this line cause it should be a new pr

**Review by dacongda — APPROVED** [review]

**Pull request merged**